src/p2p/net_node.inl

```diff
@@ -2479,6 +2479,25 @@ namespace nodetool
     std::vector<peerlist_entry> local_peerlist_new;
     zone.m_peerlist.get_peerlist_head(local_peerlist_new, true, max_peerlist_size);
 
+    /* Tor/I2P nodes receiving connections via forwarding (from tor/i2p daemon)
+    do not know the address of the connecting peer. This is relayed to them,
+    iff the node has setup an inbound hidden service. The other peer will have
+    to use the random peer_id value to link the two. My initial thought is that
+    the inbound peer should leave the other side marked as `<unknown tor host>`,
+    etc., because someone could give faulty addresses over Tor/I2P to get the
+    real peer with that identity banned/blacklisted.
+
+    \note Insert into `local_peerlist_new` so that it is only sent once like
+      the other peers. */
+    static constexpr const std::uint32_t max_random_time =
+      1 /*day*/ * 24 /*hours*/ * 60 /*minutes*/ * 60 /*seconds*/;
+    if(outgoing_to_same_zone)
+      local_peerlist_new.push_back(
+        peerlist_entry{
+          zone.m_our_address, zone.m_config.m_peer_id, std::time(nullptr) - crypto::rand_idx(max_random_time)
```

Comment:
@jeffro256 @j-berman @0xFFFC0000 

I'm looking for feedback on this `std::time(nullptr) - crypto::rand_idx(max_random_time)` call, which generates a random timestamp within the last 24 hours. The "real" peers (i.e. not our entry), should have clusters of timestamps around the graylist housekeeping callbacks.

I'm thinking about making the timestamp `0` here, and likewise putting a `0` timestamp for all peers _when not in the public zone_. I did a quick code-check, and it looks like the timestamps are **not** trusted, and therefore not really used anyway.

Thoughts?